### ISSUE TITLE
healthCheckResults: null on first task event

### ISSUE THREAD

**Issue by tysonnorris** [byxorna/marathon_http_proxy_generator#14]
due to https://github.com/mesosphere/marathon/issues/940 the behavior we see (with delay=0) is:
- task started (status == TASK_RUNNING)
- /v2/tasks includes this task in response BUT includes "healthCheckResults":null
- config is generated as if this task is healthy (UNEXPECTED)
- later, health events for this task cause /v2/tasks calls and now includes healthCheckResults": [{"alive": false}]
- config is generated as if this task is unhealthy (expected)

I'm not sure if there is anything to be done in goji about this, but it is problematic for a brief period of time where the task is referenced even though it is unhealthy.

**Comment by byxorna** [comment]
Agreed, I can see how this is a problem. I saw marathon recently introduced https://mesosphere.github.io/marathon/docs/readiness-checks.html, which is probably a direct response to the bug you mentioned :)
Unfortunately, the `/v2/tasks` endpoint doesnt mention anything about configured healthchecks with each task. We would have to query for that mapping whenever we get a task update if we wanted to know which tasks had healthchecks configured but were not reporting status. Sounds doable, but perhaps as optional behavior?

Alternatively, using readiness checks can avoid this problem entirely.
